### PR TITLE
STM32F1/F4 SoftwareSerial sanity check

### PR DESCRIPTION
**Pull request by Msq001**
### Requirements
https://github.com/teemuatlut/TMCStepper/pull/52
I create a generic SW_UART library in TMCStepper library, It comes from LPC1768 SoftwareSerial, it include STM32F1 and STM32F4 SoftwareSerial now,  if you want add a new MCU, you need only achieve `timerInit(void)`, `setSpeed(u32 speed)`, and `fastWrite`, `fastRead`

### Description

1. add SoftwareSerial for TMC2208/2209 in STM32F1/F4

### CODE REVIEW

File: Marlin/src/pins/pins_BIGTREE_SKR_PRO_V1.1.h [hunk 1]
```diff
@@ -132,31 +132,31 @@
...
 #endif
 
 //
 // Temperature Sensors
 //
-#define TEMP_0_PIN         PF3  // T0
-#define TEMP_1_PIN         PF4  // T1
-#define TEMP_2_PIN         PF5  // T2
-#define TEMP_BED_PIN       PF6  // TB
+#define TEMP_0_PIN         PF4  // T1 <-> E0
```
Comment (Msq001) [inline Marlin/src/pins/pins_BIGTREE_SKR_PRO_V1.1.h:156]: here， I compromised with the instructions. T0 for bed， T1 for E0， etc

File: Marlin/src/pins/pins_BIGTREE_SKR_PRO_V1.1.h [hunk 2]
```diff
@@ -219,7 +219,7 @@
   #endif
 
   // Alter timing for graphical display
-  #if HAS_GRAPHICAL_DISPLAY
+  //#if HAS_GRAPHICAL_DISPLAY
```
Comment (Msq001) [inline Marlin/src/pins/pins_BIGTREE_SKR_PRO_V1.1.h]: I can't found where `HAS_GRAPHICAL_DISPLAY` defined， now it will nerve be compiled

Comment (thisiskeithb) [reply Marlin/src/pins/pins_BIGTREE_SKR_PRO_V1.1.h]: It should be `HAS_GRAPHICAL_LCD` instead of `HAS_GRAPHICAL_DISPLAY`.

Comment (Msq001) [reply Marlin/src/pins/pins_BIGTREE_SKR_PRO_V1.1.h]: > It should be `HAS_GRAPHICAL_LCD` instead of `HAS_GRAPHICAL_DISPLAY`.

maybe, let me update it

**Comment by davidbilly** [comment]
@Msq001 Great, I need these fixes for TMC2209 UART. when commits going to merge? Thank you.

Edit: After I applied those changes, I faced some issue. If I use DRIVER_TYPE  TMC2209 after flashed to my SKR Pro, cant detected by my PC. 

But, TMC2209_STANDALONE able detected by my PC.

Here's my screenshot : 

TMC2209_STANDALONE  : 
skr_pro_tmc2209_standalone_01

skr_pro_tmc2209_standalone_02

skr_pro_tmc2209_standalone_03

TMC2209 : 

skr_pro_tmc2209_01

skr_pro_tmc2209_02

Please check. Thank you.

**Pull request merged**